### DISCUSSION TITLE
dis-match between xsd and field.json ?

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#336] [Q&A] [answered]
Hi all!
In notice 1 we have got : 
{
          "id" : "**BT-01-notice**",
          "contentType" : "field",
          "displayType" : "COMBOBOX",
          "description" : "Procedure Legal Basis",
          "_label" : "field|name|BT-01-notice"
        }

and 
{
          "id" : "**BT-727-Procedure**",
          "contentType" : "field",
          "displayType" : "COMBOBOX",
          "description" : "Place Performance Services Other",
          "_label" : "field|name|BT-727-Procedure"
        }

both "type" : "code"

but in UBL-CommonBasicComponents-2.3.xsd
<xsd:complexType name="RegionType">
      <xsd:simpleContent>
         **<xsd:restriction base="udt:TextType"/>**
      </xsd:simpleContent>
   </xsd:complexType>
   
and 
 <xsd:complexType name="RegulatoryDomainType">
      <xsd:simpleContent>
         **<xsd:restriction base="udt:TextType"/>**
      </xsd:simpleContent>
   </xsd:complexType>

is not it strange ?

Thanks a lot !
Renato

**Comment by YvesJo** [comment, marked as answer]
Hi Renato,
There are constraints imposed by the UBL standard that we had to deal with. This does not however prevent the use of predefined list of strings.
You could also, out of curiosity, look at [this document](https://simap.ted.europa.eu/documents/10184/320101/eForms+regulation+vs+Schemas_2/8209153c-9541-4672-9c89-25891137744c) published over 2 years ago on the [eforms dedicated page on simap](https://simap.ted.europa.eu/web/simap/eforms)
KR,
Yves

**Reply by rengit62** [reply]
Hi Yves,
just take in consideration BT-727-Procedure. I guess that is defined code in fields.json : 
"id" : "BT-727-Procedure",
    "parentNodeId" : "ND-ProcedurePlacePerformance",
    "name" : "Place Performance Services Other",
    "btId" : "BT-727",
    "xpathAbsolute" : "/*/cac:ProcurementProject/cac:RealizedLocation/cac:Address/cbc:Region",
    "xpathRelative" : "cbc:Region",
    "xsdSequenceOrder" : [ { "cbc:Region" : 24 } ],
    "type" : "code",
    "legalType" : "CODE",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "T01", "T02", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      } ]
    },
    "codeList" : {
      "value" : {
        "id" : "other-place-service",
        "type" : "flat"
      },
      "severity" : "ERROR"
    }

and COMBOX in <noticeSubtype>.json  : 
"id" : "BT-727-Procedure",
          "contentType" : "field",
          "displayType" : "COMBOBOX",
          "description" : "Place Performance Services Other",
          "_label" : "field|name|BT-727-Procedure"

so that in the form so as to become a SELECT HTML 

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/35107d98-07fd-4857-a9b1-9b9360311371)

but when we build the xml file as the field has got : 
"type" : "code",
    "legalType" : "CODE", and codeList" it should have the attribute @listName as you wrote in doc in the DynamicProperties  : 
https://docs.ted.europa.eu/eforms/latest/fields/index.html#field-repository

But if we put the attribute @listName we get an xsd error because TextType in xsd has got 2 optional attributes : 
languageID and languageLocaleID
So 2 questions : 
1) Is there a rule, any attribute, .. by which we can exclude BT-727-Procedure to have @ListName attribute ?
Of course we can set an 'ugly if ' in the code but this is opposite to the concept of a metadata driven application.
2) if we would se one of the 2 attribute to <cbc:Region>anyw-cou</cbc:Region> which of them should be set ?
Thanks and Regards
Renato

**Reply by YvesJo** [reply]
Hi Renato,
This is a known issue that you also already reported.
Systematically adding a listName attribute may lead to an error (e.g. when the XML element is of type _text_ or _identifier_ and not of type _code_ itself) or not make sense at all when the field already points to an attribute (e.g. BT-03-notice).
We know that having to analyze the schema is not a solution and we are looking for an alternate option.
The use of attributes for other purpose than the one they have been defined for is not recommended. In the case of a text element containing a value from a code list, no attribute shall be used.
KR,
Yves
